### PR TITLE
U610-067 Pretty printer options

### PR DESCRIPTION
**Pull request by ogorodnik**
Using the corresponding project that a file belongs to
 get the options.

### CODE REVIEW

File: source/ada/lsp-ada_contexts.adb [hunk 1]
```diff
@@ -814,6 +766,67 @@ package body LSP.Ada_Contexts is
...
+      Project.Switches
+        (In_Pkg           => "Pretty_Printer",
+         File             => File,
+         Language         => "ada",
+         Value            => Options,
+         Is_Default_Value => Default);
+
+      --  Initialize an gnatpp command line object
+      Last := Options'First - 1;
+      for Item of Options.all loop
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_contexts.adb:796]: Lacking comments. Put a comment to say that we are counting valid options here for instance, people don't want to analyze each loop in your code

File: source/ada/lsp-ada_contexts.adb [hunk 2]
```diff
@@ -814,6 +766,67 @@ package body LSP.Ada_Contexts is
...
+         if Item /= null
+           and then Item.all /= ""
+         then
+            Last := Last + 1;
+         end if;
+      end loop;
+
+      Validated := new GNAT.Strings.String_List (Options'First .. Last);
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_contexts.adb:804]: "Validated_Options" would be even better

File: source/ada/lsp-ada_contexts.ads [hunk 3]
```diff
@@ -257,17 +257,19 @@ private
...
-      --  Object to keep gnatpp options
-
       Follow_Symlinks : Boolean := True;
       --  See LSP.Ada_Handlers for description
 
       Reader_Reference : Langkit_Support.File_Readers.File_Reader_Reference;
       --  A reference to the file reader created for this context
    end record;
 
+   procedure Get_PP_Options
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_contexts.ads:267]: Comment

File: source/ada/lsp-ada_handlers.adb [hunk 4]
```diff
@@ -247,6 +247,13 @@ package body LSP.Ada_Handlers is
       URI  : LSP.Types.LSP_String) return LSP.Types.LSP_String;
    --  Turn URI into path
 
+   function Get_Project
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_handlers.adb:250]: LSP.Ada_Documents seems a better package to put that function (might be reused somewhere else in the future)

File: source/ada/lsp-ada_handlers.adb [hunk 5]
```diff
@@ -247,6 +247,13 @@ package body LSP.Ada_Handlers is
...
+     (Self : access Message_Handler;
+      Uri  : LSP.Messages.DocumentUri)
+      return GNATCOLL.Projects.Project_Type;
+   --  Retrieve the project that the file belongs to. Root project is returned
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_handlers.adb:254]: You should say what it does for aggregate projects (that's the purpose of your function)

**Review by AnthonyLeonardoGracio — CHANGES_REQUESTED** [review]

**Comment by reznikmm** [comment]
closing till we get real issue with this

**Pull request closed without merging**